lineapy/graph_reader/artifact_collection.py

```diff
@@ -175,24 +188,68 @@ def _sort_session_artifacts(
                 + ", ".join(list(combined_taskgraph.nodes))
             )
             raise KeyError(msg)
-        combined_taskgraph.add_edges_from(task_dependency_edges)
+
         # Check if the graph is acyclic
         if nx.is_directed_acyclic_graph(combined_taskgraph) is False:
             raise Exception(
                 "LineaPy detected conflict with the provided dependencies. "
                 "Please check if the provided dependencies include circular relationships."
             )
 
-        # Construct inter session dependency graph
+    def create_inter_session_graph(self, dependencies: TaskGraphEdge = {}):
```

Comment:
Maybe call this `create_inter_session_taskgraph` to make sure this is a TaskGraph not a LineaGraph

Reply:
renamed, this is a good clarification, thanks for catching this.